shared/speaker_setups.hpp

```diff
@@ -0,0 +1,441 @@
+      Speaker{"AC_00010015", "BtFC", "B+000", 0.f, -30.f, Layer::bottom, false, false},
+      Speaker{"AC_00010016", "BtFL", "B+045", 45.f, -30.f, Layer::bottom, false, false},
+      Speaker{"AC_00010017", "BtFR", "B-045", -45.f, -30.f, Layer::bottom, false,false}
+    }, 8),
+
+  SpeakerSetup(
+    "9+9+0", "Auro-3D", "BS.2094", "Auro-3D", "AP_00010011",
+    std::vector<Speaker>{
+      Speaker{"AC_00010001", "L", "M+030", 30.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_00010002", "R", "M-030", -30.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_00010003", "C", "M+000", 0.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_00010004", "LFE", "LFE1", 45.f, -30.f, Layer::bottom, true, false},
+      Speaker{"AC_00010005", "Ls", "M+110", 110.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_00010006", "Rs", "M-110", -110.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_0001000a", "Lss", "M+090", 90.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_0001000b", "Rss", "M-090", -90.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_0001001a", "BL", "M+135", 135.f, 0.f, Layer::middle, false, false}, // This speaker is marked as diffuse
+      Speaker{"AC_0001001b", "BR", "M-135", -135.f, 0.f, Layer::middle, false, false},  // This speaker is marked as diffuse
+      Speaker{"AC_0001000d", "Ltf", "U+030", 30.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_0001000f", "Rtf", "U-030", -30.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_0001000e", "TpFC", "U+000", 0.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010010", "Ltr", "U+110", 110.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010012", "Rtr", "U-110", -110.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010013", "TpSiL", "U+090", 90.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010014", "TpSiR", "U-090", -90.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_0001001e", "Ltb", "U+135", 135.f, 30.f, Layer::upper, false, false},
+      Speaker{"AC_0001001f", "Rtb", "U-135", -135.f, 30.f, Layer::upper, false,false}
+    }),
+
+  /* clang-format on */
+};
+
+inline SpeakerSetup speakerSetupByName(const std::string& name) {
+  auto it = std::find_if(
+      SPEAKER_SETUPS.begin(), SPEAKER_SETUPS.end(),
+      [name](auto speakerSetup) { return speakerSetup.name == name; });
+  if (it != SPEAKER_SETUPS.end()) {
+    return *it;
+  } else {
+    return SpeakerSetup{};
+  }
+}
+
+inline SpeakerSetup speakerSetupByCommonName(const std::string& commonName) {
+  auto it = std::find_if(SPEAKER_SETUPS.begin(), SPEAKER_SETUPS.end(),
+                         [commonName](auto speakerSetup) {
+                           return speakerSetup.commonName == commonName;
+                         });
+  if (it != SPEAKER_SETUPS.end()) {
+    return *it;
+  } else {
+    return SpeakerSetup{};
+  }
+}
+
+inline SpeakerSetup speakerSetupByIndex(int index) {
```

Comment:
Does this still get used anywhere? It feels like we should either be looking up by pack format id or legacy index, does having this in as well break encapsulation / provide an easy way to look up things incorrectly?